docs/tools/CLI/build_profiles.md

```diff
@@ -1,42 +1,42 @@
+Arm Mbed OS 5 defines three collections of toolchain flags used during the build. These are __build profiles__. These are collections of toolchain flags used during the build. The three build profiles are *develop*, *debug* and *release*. The Mbed Online Compiler uses the *develop* build profile. When building from Arm Mbed CLI, you may select the __build profile__ by passing your desired build profile, by name or path, to the __--profile__ argument.
 
-### Develop profile
+### Develop
 
 - Small and fast code.
 - Full error information. For example, asserts have file name and line number.
 - Hard to follow code flow when using a debugger.
 - Chip goes to sleep when idle:
-    * Debugger is likely to drop connection.
-    * Breaks the local file system on the [Arm Mbed interface](../introduction/index.html) on some boards.
+   - Debugger is likely to drop connection.
+   - Breaks the local file system on the [Arm Mbed interface](../introduction/index.html) on some boards.
 
-### Debug profile
+### Debug
 
-- Largest and slowest profile.
+- Largest and slowest performance.
 - Full error information. For example, asserts have file name and line number.
 - Easy to step through code with a debugger.
 - Disabled sleep mode.
 
-### Release profile
+### Release
 
-- Smallest profile and still fast.
+- Smallest code size and still fast.
 - Minimal error information.
 - Chip goes to sleep when going idle:
-    - Debugger is likely to drop connection.
-    - Breaks the local file system on the [Mbed interface](../introduction/index.html) on some boards.
+   - Debugger is likely to drop connection.
+   - Breaks the local file system on the [Mbed interface](../introduction/index.html) on some boards.
 
-### User defined profiles
+### User-defined build profile
 
-A toolchain or build system profile is a set of flags that is guaranteed to be passed to the underlying compiler suite.
+As mentioned above, the __build profile__ defines the set of flags that is guaranteed to be passed to the underlying compiler suite. You can also create custom or user-defined __build profile__ using a JSON file acording to the __build profile__ format mentioned in [JSON build profile format](#JSON-build-profile-format).
 
-These flags are stored in a JSON file that may be merged with other JSON files of the same structure.
+These flags stored in the JSON file may also be merged with other JSON files of the same structure.
 
-#### JSON toolchain profile format
+#### JSON build profile format
 
-The JSON object that represents a toolchain profile is a dictionary mapping from toolchains, such as `GCC_ARM`, to their flags, such as `-O3`.
+The __build profiles__ are implemented as JSON files with the root object representing toolchain configurations for mapping each supported toolchains, such as `GCC_ARM`, to their flags, such as `-O3`.
```

Comment:
Build profiles are not implemented. Build profile _are_ JSON files. What does "toolchain configurations mean"? can we just say that they are a JSON object with a key for each toolchain?
```suggestion
The __build profiles__ are JSON files with a root object representing toolchain configurations for mapping each supported toolchains, such as `GCC_ARM`, to their flags, such as `-O3`.
```

or 

```suggestion
__build profiles__ are JSON files that map each supported toolchain, such as `GCC_ARM`, to their flags, such as `-O3`.
```